### PR TITLE
[feat-75]: Email verification

### PR DESCRIPTION
**Pull request by bek-shoyatbek**
[Screencast from 2024-12-17 21-43-41.webm](https://github.com/user-attachments/assets/f0925a82-b377-40ce-9b91-7383784faae1)

### CODE REVIEW

File: src/modules/auth/components/email-verification/check-mail-box.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import EmailOutlinedIcon from '@mui/icons-material/EmailOutlined';
+import { checkMailBoxStyles } from './check-mail-box.styles';
+
+export function CheckMailbox({ email }: { email: string }) {
+  return (
+    <Container maxWidth="xs" sx={checkMailBoxStyles.container}>
+      <Paper elevation={3} sx={checkMailBoxStyles.paper}>
+        <EmailOutlinedIcon sx={checkMailBoxStyles.icon} />
+        <Typography variant="h5" gutterBottom={true}>
+          Verify Your Email
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/check-mail-box.tsx:13]: add translation for static texts

File: src/modules/auth/components/email-verification/email-verification.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  };
+
+  const renderContent = () => {
+    switch (status) {
+      case 'loading':
+        return (
+          <>
+            <CircularProgress />
+            <Typography variant="h6" style={{ marginTop: '20px' }}>
+              Verifying your email...
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/email-verification.tsx:58]: for this also

File: src/modules/auth/components/email-verification/email-verification.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            </Typography>
+          </>
+        );
+      case 'success':
+        return (
+          <>
+            <CheckCircleOutlineIcon style={{ ...emailVerificationStyles.icon, color: 'green' }} />
+            <Typography variant="h5" gutterBottom={true}>
+              Email Verified Successfully
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/email-verification.tsx:67]: for this also

File: src/modules/auth/components/email-verification/email-verification.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        );
+      case 'error':
+        return (
+          <>
+            <ErrorOutlineIcon style={{ ...emailVerificationStyles.icon, color: 'red' }} />
+            <Typography variant="h5" gutterBottom={true}>
+              {errorMsg}
+            </Typography>
+            <Typography variant="body1" color="textSecondary">
+              There was an issue verifying your email. The link may be invalid or expired.
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/email-verification.tsx:79]: for this also

File: src/modules/auth/components/email-verification/email-verification.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+              variant="outlined"
+              color="primary"
+              style={emailVerificationStyles.button}
+              onClick={() => {
+                (async () => {
+                  await handleResendVerification();
+                })();
+              }}
+            >
+              Resend Verification Email
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/email-verification.tsx:91]: for this also

File: src/modules/auth/components/email-verification/check-mail-box.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        </Typography>
+        <Typography variant="body1" color="textSecondary">
+          We have sent a verification email to
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/check-mail-box.tsx:16]: for this also

File: src/modules/auth/components/email-verification/check-mail-box.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        </Typography>
+        <Typography variant="subtitle1" color="primary">
+          {email}
+        </Typography>
+        <Typography variant="body2" color="textSecondary">
+          Please check your inbox and click the verification link to activate your account.
```
Comment (purple-mountain) [inline src/modules/auth/components/email-verification/check-mail-box.tsx:22]: for this also

File: src/modules/auth/components/email-verification/check-mail-box.tsx (hunk 1 above)
Comment (bek-shoyatbek) [reply src/modules/auth/components/email-verification/check-mail-box.tsx:13]: where can I add it?

Comment (purple-mountain) [reply src/modules/auth/components/email-verification/check-mail-box.tsx:13]: public json file for English and Russian and translation keys constant for referencing json.

 you can use useTranslation hook. 

for reference you can see how I did it in auth or in any other pages.

First write English one then you can use chatgpt for generating Russian json based on English json

**Review by purple-mountain — APPROVED** [review]

**Pull request merged**